public/app/features/alerting/unified/state/AlertingQueryRunner.ts

```diff
@@ -81,15 +104,26 @@ export class AlertingQueryRunner {
   // to do this we will convert the list of queries into a DAG and walk the invalid node's output edges recursively
   async prepareQueries(queries: AlertQuery[]) {
     const queriesToExclude: string[] = [];
-
-    // convert our list of queries to a graph
-    const queriesGraph = createDagFromQueries(queries);
+    let queriesGraph: Graph | undefined;
+
+    // exclude nodes that failed to link
+    try {
+      queriesGraph = createDagFromQueries(queries);
+    } catch (error) {
+      if (isDagError(error)) {
```

Comment:
nit: maybe we should add custom errors like `DagError`, and `DagLinkingError`? Then typecheking would become trivial as we could just use `error instanceof DagError`

Reply:
Yep, that seems about right for what we're trying to do